hnswlib/hnswlib.h

```diff
@@ -60,8 +77,8 @@ namespace hnswlib {
     template<typename dist_t>
     class AlgorithmInterface {
     public:
-        virtual void addPoint(void *datapoint, labeltype label)=0;
-        virtual std::priority_queue<std::pair<dist_t, labeltype >> searchKnn(const void *, size_t) const = 0;
+        virtual void addPoint(const void *datapoint, labeltype label)=0;
+        virtual retType<dist_t> searchKnn(const void *, size_t) const = 0;
```

Comment:
Changing the result of the `searchKnn` breaks backward compatibility with all C++ code...
I think adding a separate method with template comparator would be a better solution, isn't it?